src/main/java/org/commcare/core/interfaces/VirtualDataInstanceCache.java

```diff
@@ -0,0 +1,20 @@
+package org.commcare.core.interfaces;
+
+import org.javarosa.core.model.instance.VirtualDataInstance;
+
+import java.sql.SQLException;
+import java.util.UUID;
+
+import javax.annotation.Nullable;
+
+/**
+ * Read and write operations for entity selections made on a mult-select Entity Screen
+ */
+public interface VirtualDataInstanceCache {
```

Comment:
how about renaming this to `VirtualDataInstanceStorage` since it is a primary store and not a cache?

Reply:
Agreed. I think 'cache' sets the wrong expectation for how this is used.

Reply:
Implemented in the other PR to avoid conflicts - https://github.com/dimagi/commcare-core/pull/1109/commits/5ebe896802bc4a4b0e7e3b4edcaf5d5b8d8c566b